### PR TITLE
#3619 Additional name fields

### LINKED ISSUE #3619: Adding additional fields to the name form
## Is your feature request related to a problem? Please describe.

We need to add some additional fields to a name record which require input widgets to be added to the patient form.

## Describe the solution you'd like

For all of the fields here: https://github.com/openmsupply/mobile/issues/3614
- Add a `FormInputConfig` here: https://github.com/openmsupply/mobile/blob/471c1e79f1d2aec239401e9e45c6ee8eea2295c3/src/utilities/formInputConfigs.js#L64
- Add the input configs to the patient form config: https://github.com/openmsupply/mobile/blob/471c1e79f1d2aec239401e9e45c6ee8eea2295c3/src/utilities/formInputConfigs.js#L313-L323

?

## Implementation

As above

## Describe alternatives you've considered

Doing the patient form the same as the new patient survey form 🤷 

## Additional context

Depending on how the form survey is implemented, the patient form might look funny!

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #3619 

mSupply issue with footrunner code which can be run to generate required data https://github.com/sussol/msupply/pull/8196

## Change summary

- Adds two new conditional fields to the patient edit form control- nationality/citizenship and ethnicity

## Testing

- [ ] Check patient form in dispensary module
- [ ] Check patient form in vaccine dispensary
- [ ] Check other drop downs are still OK in dispensary module and cash register (NOTE: To enable cash register, you must have patient types enabled..?)

### Related areas to think about

- Didn't put occupation. We need employer also, so thought to put occupation in the survey form
- Didn't put village. Dunno how we'd get that list, so we can either offer them to put it in one of the address fields or add it to the survey form.

### CODE REVIEW

**Review by mark-prins — COMMENTED** [review]
The `Name.toJSON` needs to have `ethnicity` and `nationality` added for the dispensing form to pick up the values

File: src/utilities/formInputConfigs.js [hunk 1]
```diff
@@ -206,6 +208,26 @@ const FORM_INPUT_CONFIGS = seedObject => ({
     label: formInputStrings.family_policy_number,
     isEditable: !seedObject,
   },
+  [FORM_INPUT_KEYS.NATIONALITY]: {
+    type: FORM_INPUT_TYPES.DROPDOWN,
+    initialValue: seedObject?.nationality ?? null,
+    key: 'nationality',
+    label: formInputStrings.citizenship,
+    options: UIDatabase.objects('Nationality'),
```
[the remark this answers is not in the document]
Comment (mark-prins) [reply src/utilities/formInputConfigs.js:216]: on second thoughts.. this is the order provided and I would have imported the same way.
Let's do as instructed and keep Tonga up the top

**Comment by joshxg** [comment]
oops! 
818018d

Re order: I had thought to put a `order` int on the tables which could use `-1` as the default, but basically went in the too hard basket...

**Review by mark-prins — APPROVED** [review]
On second thoughts I think the order is fine. it shows up how it was requested, so that seems ok to me

**Pull request merged**